genai-perf/genai_perf/telemetry_data/triton_telemetry_data_collector.py

```diff
@@ -66,34 +66,56 @@ def _process_and_update_metrics(self, metrics_data: str) -> None:
             ```
 
             The method will extract and process:
-            - `nv_gpu_power_usage` as `gpu_power_usage`
-            - `nv_gpu_utilization` as `gpu_utilization`
-            - `nv_energy_consumption` as `energy_consumption`
+            'gpu_power_usage': {
+                'gpu0': [27.01]
+            },
+            'gpu_utilization': {
+                'gpu0': [75.5]
+            },
+            'energy_consumption': {
+                'gpu0': [1234.56]
+            }
         """
 
         if not metrics_data.strip():
             logger.info("Response from Triton metrics endpoint is empty")
             return
 
+        gpu_mapping = {}
+        gpu_index = 0
+
         current_measurement_interval = {
-            metric.name: [] for metric in self.metrics.TELEMETRY_METRICS
-        }  # type: Dict[str, List[float]]
+            metric.name: {} for metric in self.metrics.TELEMETRY_METRICS
+        }  # type: Dict[str, Dict[str, List[float]]]
 
         for line in metrics_data.splitlines():
             line = line.strip()
-            if not line:
+            if not line or line.startswith("#"):
```

Comment:
why was this check added?
does the metrics_data include comments?

Reply:
Yep, it has comments.
Below is the a part of metrics from Triton 
```
# HELP nv_inference_request_success Number of successful inference requests, all batch sizes
# TYPE nv_inference_request_success counter
nv_inference_request_success{model="vllm_model",version="1"} 7918
# HELP nv_inference_request_failure Number of failed inference requests, all batch sizes
# TYPE nv_inference_request_failure counter
nv_inference_request_failure{model="vllm_model",reason="BACKEND",version="1"} 0
nv_inference_request_failure{model="vllm_model",reason="CANCELED",version="1"} 0
nv_inference_request_failure{model="vllm_model",reason="OTHER",version="1"} 0
nv_inference_request_failure{model="vllm_model",reason="REJECTED",version="1"} 0
# HELP nv_inference_count Number of inferences performed (does not include cached requests)
# TYPE nv_inference_count counter
nv_inference_count{model="vllm_model",version="1"} 7918
# HELP nv_inference_exec_count Number of model executions performed (does not include cached requests)
# TYPE nv_inference_exec_count counter
nv_inference_exec_count{model="vllm_model",version="1"} 7918
# HELP nv_inference_request_duration_us Cumulative inference request duration in microseconds (includes cached requests)
# TYPE nv_inference_request_duration_us counter
nv_inference_request_duration_us{model="vllm_model",version="1"} 452661337
# HELP nv_inference_queue_duration_us Cumulative inference queuing duration in microseconds (includes cached requests)
# TYPE nv_inference_queue_duration_us counter
nv_inference_queue_duration_us{model="vllm_model",version="1"} 128932
# HELP nv_inference_compute_input_duration_us Cumulative compute input duration in microseconds (does not include cached requests)
# TYPE nv_inference_compute_input_duration_us counter
nv_inference_compute_input_duration_us{model="vllm_model",version="1"} 192290
# HELP nv_inference_compute_infer_duration_us Cumulative compute inference duration in microseconds (does not include cached requests)
# TYPE nv_inference_compute_infer_duration_us counter
nv_inference_compute_infer_duration_us{model="vllm_model",version="1"} 452294570
# HELP nv_inference_compute_output_duration_us Cumulative inference compute output duration in microseconds (does not include cached requests)
```